Fi 2727 inferno ci disable tls

Shaumik-Ashraf opened the pull request:
# Summary

Forked this branch from #167, but am using an experimental fork of g10 test kit.

## New behavior

Add `INFERNO_DISABLE_TLS_TEST` to env, which causes the TLS test to omit itself. Rollup should pass, and all groups should be able to run in 1 test run allowing scratch to work as expected. This should resolve issue FI-3367 [discussed](https://github.com/inferno-framework/inferno-reference-server/pull/180#issuecomment-2436508631) in PR #180. 

## Code changes

## Testing guidance

Observe Github Actions

Shaumik-Ashraf commented on .github/workflows/inferno_ci.yml:
```diff
@@ -0,0 +1,124 @@
+                                               additional_patient_ids:85,355 \
+                                               "smart_credentials:{\"access_token\":\"SAMPLE_TOKEN\"}"
+
+      - name: Run g10 test kit Multi-Patient API with US Core 3.1.1 and Bulk Data 1.0.1
```
It can work, I just thought this way was more legible.

Shaumik-Ashraf replied:
Done!

dehall commented on .github/workflows/inferno_ci.yml:
```diff
@@ -0,0 +1,124 @@
+                                               "since_timestamp:2024-10-10T23:52:09+00:00" \
+                                               "smart_credentials:{\"access_token\":\"SAMPLE_TOKEN\"}"
+
+      - name: Stop server
```
Are these last 2 strictly necessary for anything or are they just a demonstration of cleaning up nicely?
If they are necessary you might want a `if: always()` to ensure they run even if a test fails.

dehall commented on .github/workflows/inferno_ci.yml:
```diff
@@ -0,0 +1,124 @@
+env:
+  TEST_KIT_REPO: https://github.com/Shaumik-Ashraf/onc-certification-g10-test-kit.git
+  TEST_KIT_NAME: onc-certification-g10-test-kit
+  TEST_KIT_RELEASE: fi-3367-disable-tls-for-ci
```
I still really don't like using a fixed version here. Every time we release a new version of g10 this means we'll have another manual step of updating this. I don't think we'll intentionally want to test against an older version. (unlike something like the ruby version where a repo might want to support a range) 
We don't necessarily want to run against the main branch, sure, but "the latest release" would be ideal.

Shaumik-Ashraf commented on .github/workflows/inferno_ci.yml:
```diff
@@ -0,0 +1,124 @@
+                                               bulk_patient_ids_in_group:85,355 \
+                                               "smart_credentials:{\"access_token\":\"SAMPLE_TOKEN\"}"
+
+      - name: Run g10 test kit Single Patient API with US Core 6.1.0
```
I agree, but doing this as a matrix is another ticket (FI-3373)

arscan replied:
Please update that ticket so that it isn't just about doing more combos but it is also about not halting on the first version that fails, if that is the case here.  I didn't realize that is how this works.

arscan commented:
> Ready for re-review and merge!

Could you just quickly reply with what you did (if anything) in the above comments @Shaumik-Ashraf to expedite review?

Shaumik-Ashraf commented:
> > Ready for re-review and merge!
> 
> Could you just quickly reply with what you did (if anything) in the above comment threads @Shaumik-Ashraf to expedite review?

I'll summarize here:
 - I showcase `inferno execute` running multiple groups together by putting the Single Patient API and Multi-Patient API steps together
 - I have the CI always use the latest release of (g)(10), but I also left commented out examples of using a Test Kit's fixed release and main branch.
 - Just now I fixed and cleaned up my remaining comments. I'm trying to keep this workflow file dev friendly. Thanks @dehall for the catch
